[testing123456789101121314] iP

testing1234567891011121314 opened the pull request:
# Reminderebot
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Reminderebot frees your mind of having to remember things you need to do. It's,

+ text-based
+ easy to learn
+ ~~FAST~~ _SUPER_ FAST to use
+ All you need to do is,

download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
1. double-click it.
2. __add__ your tasks.
3. let it __manage your tasks__ for you 😉
4. And it is __FREE__!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Use `git status` to list all new or modified files that haven't yet been committed.
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

testing1234567891011121314 committed:
Level 0

testing1234567891011121314 committed:
Add Level 0

Reminderebot class:
rename Duke.java
simply greets user and exits

testing1234567891011121314 committed:
Delete class file
Reminderebot.class: Remove file

testing1234567891011121314 committed:
Add Level 1
Edit Reminderebot class:
Echos commands entered by the user, and exits when the user types the command bye

testing1234567891011121314 committed:
Add Level 1
Edit Reminderebot class:
Echos commands entered by the user, and exits when the user types the command bye

testing1234567891011121314 committed:
Revert "Add Level 1"

This reverts commit 8c40f057d4049b8f74cf4b3d3ae08562b3e4686b.

testing1234567891011121314 committed:
Merge branch 'master' of https://github.com/testing1234567891011121314/ip

testing1234567891011121314 committed:
bug fix: remove space in class name
Reminderebot class: edit class name

testing1234567891011121314 committed:
Add Level 2
Reminderebot class: Add the ability to store whatever text entered by the user and display them back to the user when requested
add printTasks() for showing tasklist
add addTasks() for saving tasklist

testing1234567891011121314 committed:
Add Level 3
Class Task:
handles mark and unmark
Class Reminderebot:
Add the ability to store whatever text entered by the user and display them back to the user when requested

testing1234567891011121314 committed:
Add Level 4
Class Reminderebot: Add support for tracking three types of tasks
Class ToDo: tasks without any date/time attached to it e.g., visit new theme park
Class Deadline: tasks that need to be done before a specific date/time e.g., submit report by 11/10/2019 5pm
Class Event: tasks that start at a specific date/time and ends at a specific date/time
e.g., (a) team project meeting 2/10/2019 2-4pm (b) orientation week 4/10/2019 to 11/10/2019

testing1234567891011121314 committed:
Add A-TextUiTesting
runtest.bat: runs test on class Reminderebot instead of Duke
input.txt: provides input for test cases

testing1234567891011121314 committed:
Add A-Exception
Class ReminderebotException: For all ReminderebotException

testing1234567891011121314 committed:
Add A-Exception
Class ReminderebotException: For all ReminderebotException

testing1234567891011121314 committed:
Merge branch 'master' of https://github.com/testing1234567891011121314/ip

testing1234567891011121314 committed:
Add Level-5
ReminderebotException: handles invalid inputs for mark, unmark, todo, deadline and event

testing1234567891011121314 committed:
Add Level 6
Reminderebot: deletion of tasks implemented
Commands: enum seperated from Reminderebot
Task: set as abstract class
EXPECTED.TXT: updated for regression testing

testing1234567891011121314 committed:
Add Level-7 in progress
Class TaskList: handles creation and management of task list
Class Storage: handles writing to file
Class Task: support conversion to String for storage
Class ToDo: support conversion to String for storage
Class Deadline: support conversion to String for storage
Class Event: support conversion to String for storage
Enum Commands: added support for Delete

testing1234567891011121314 committed:
Add Level-7 in progress
Class TaskList: supports getting task and getting length
Class Storage: handles writing to file
Class Task: support conversion to String for storage
Class ToDo: support conversion to String for storage
Class Deadline: support conversion to String for storage
Class Event: support conversion to String for storage

testing1234567891011121314 committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Add Level-7 in progress Class TaskList: supports getting task and getting length Class Storage: handles writing to file Class Task: support conversion to String for storage Class ToDo: support conversion to String for storage Class Deadline: support conversion to String for storage Class Event: support conversion to String for storage
  Add Level-7 in progress Class TaskList: handles creation and management of task list Class Storage: handles writing to file Class Task: support conversion to String for storage Class ToDo: support conversion to String for storage Class Deadline: support conversion to String for storage Class Event: support conversion to String for storage Enum Commands: added support for Delete

testing1234567891011121314 committed:
Add Level-8
Class Reminderebot: handle datetime of format "dd/MM/yy HHmm" and display them with format "MMM d yyyy hhmm a"
Storage: handle conversion of string to datetime
Task: handle display of datetime
Deadline: support handling and displaying of datetime
Event: support handling and displaying of datetime

testing1234567891011121314 committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Add Level-8 Class Reminderebot: handle datetime of format "dd/MM/yy HHmm" and display them with format "MMM d yyyy hhmm a" Storage: handle conversion of string to datetime Task: handle display of datetime Deadline: support handling and displaying of datetime Event: support handling and displaying of datetime

testing1234567891011121314 committed:
Add A-MoreOOP
Seperate code into Classes Ui, Parser, Storage, Tasklist
Abstract class Command: represent a command for Reminderebot

testing1234567891011121314 committed:
Add A-MoreOOP
Implement Command.java with all commands

testing1234567891011121314 committed:
Merge branch 'branch-A-MoreOOP'

* branch-A-MoreOOP:
  Add A-MoreOOP Implement Command.java with all commands
  Add A-MoreOOP Seperate code into Classes Ui, Parser, Storage, Tasklist Abstract class Command: represent a command for Reminderebot

testing1234567891011121314 committed:
Add A-Packages
package reminderebot
package reminderebot.commands
package reminderebot.task

testing1234567891011121314 committed:
Merge branch 'branch-A-Packages'

* branch-A-Packages:
  Add A-Packages package reminderebot package reminderebot.commands package reminderebot.task

testing1234567891011121314 committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

testing1234567891011121314 committed:
Add A-Gradle
build.gradle: edit mainClass to reminderebot

testing1234567891011121314 committed:
Add A-JUnit
ParserTest: Test if the parse method works as intended
TaskListTest: Test if the delete method works as intended
ToDoTest: Test if toFile method work as intended

testing1234567891011121314 committed:
Merge branch 'branch-A-JUnit'

* branch-A-JUnit:
  Add A-JUnit ParserTest: Test if the parse method works as intended TaskListTest: Test if the delete method works as intended ToDoTest: Test if toFile method work as intended

testing1234567891011121314 committed:
Add A-JavaDoc
JavaDoc for all classes and methods

testing1234567891011121314 committed:
Add A-CodingStandard
Standardise code to follow coding standard

testing1234567891011121314 committed:
Add Level-9
Enum Commands: add Find command
Class Parser: add support for Find command
Class Ui: add support to print tasks found
Class Task: add support for Find command

testing1234567891011121314 committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add A-JavaDoc JavaDoc for all classes and methods

testing1234567891011121314 committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Add A-CodingStandard Standardise code to follow coding standard

testing1234567891011121314 committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add Level-9 Enum Commands: add Find command Class Parser: add support for Find command Class Ui: add support to print tasks found Class Task: add support for Find command

# Conflicts:
#	src/main/java/reminderebot/Ui.java

ckclion commented on src/main/java/reminderebot/Reminderebot.java:
```diff
@@ -0,0 +1,61 @@
+public class Reminderebot {
+    // note: needs support for handling multi-line inputs
+    private Storage storage;
+    private TaskList tasklist;
```
Perhaps plural form like "tasks" would be better here?

testing1234567891011121314 replied:
Thank you for the suggestion!

ckclion commented on src/main/java/reminderebot/Storage.java:
```diff
@@ -0,0 +1,146 @@
+        //  System.out.println(Arrays.toString(fields));  // debug
+        Task taskToAdd;
+
+        switch (fields[0]) {
```
Good job for noticing the switch case indentation 👍

ckclion commented on src/main/java/reminderebot/Parser.java:
```diff
@@ -0,0 +1,122 @@
+package reminderebot;
+
+import reminderebot.commands.*;
```
Perhaps imports should be listed explicitly

ckclion commented on src/main/java/reminderebot/Storage.java:
```diff
@@ -0,0 +1,146 @@
+import reminderebot.task.ToDo;
+import reminderebot.task.Deadline;
+import reminderebot.task.Event;
+import java.io.*;
```
Same issue as mentioned

ckclion commented on src/main/java/reminderebot/task/Event.java:
```diff
@@ -0,0 +1,49 @@
+     */
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (from: " + from.format(formatter) +
+                " to: " + to.format(formatter) + ")";
```
Perhaps you should break before an operator

ckclion commented on src/main/java/reminderebot/TaskList.java:
```diff
@@ -0,0 +1,87 @@
+package reminderebot;
```
I like that every class is in a package

ckclion reviewed (commented):
LGTM. Just some nits to fix.

testing1234567891011121314 committed:
Add A-Checkstyle
build.gradle: add checkstyle
checkstyle.xml: check if code is standard
Class Storage: add reference
Class Ui: deconflict merge

testing1234567891011121314 committed:
Add A-Checkstyle
build.gradle: add checkstyle
checkstyle.xml: check if code is standard
Class Storage: add reference
Class Ui: deconflict merge

testing1234567891011121314 committed:
Merge branch 'branch-A-CheckStyle' of https://github.com/testing1234567891011121314/ip into branch-A-CheckStyle

testing1234567891011121314 committed:
Merge branch 'branch-A-CheckStyle' of https://github.com/testing1234567891011121314/ip into branch-A-CheckStyle

testing1234567891011121314 committed:
Merge branch 'branch-A-CheckStyle' of https://github.com/testing1234567891011121314/ip into branch-A-CheckStyle

* 'branch-A-CheckStyle' of https://github.com/testing1234567891011121314/ip:

testing1234567891011121314 committed:
Merge branch 'branch-A-CheckStyle'

* branch-A-CheckStyle:
  Add A-Checkstyle build.gradle: add checkstyle checkstyle.xml: check if code is standard Class Storage: add reference Class Ui: deconflict merge
  Add A-Checkstyle build.gradle: add checkstyle checkstyle.xml: check if code is standard Class Storage: add reference Class Ui: deconflict merge

testing1234567891011121314 committed:
Add Level-10
Class Reminderebot: first iteration
Class Launcher: Launch GUI
Class Dialogbox: styles dialogbox for GUI
Class Main: Build application
Add images for user and Reminderebot
Add controls for dialogbox

testing1234567891011121314 committed:
Add Level-10
Class Reminderebot: second iteration
Class Main: second iteration

testing1234567891011121314 committed:
Add Level-10
Class DialogBox: third iteration
Class Main: third iteration

waihin26 commented on src/main/java/reminderebot/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,52 @@
+     */
+    @Override
+    public boolean isExit() {
+        return false;
```
Overall the code was quite well written. Nice use of OOP principles here that led to it being easy to understand.

waihin26 reviewed (commented):
Well written code

testing1234567891011121314 committed:
Add Level-10
Class Command: Support GUI
Class Main: Support FXML
Class Reminerebot: Support GUI
Class TaskList: Support GUI
Class Ui: Support GUI

testing1234567891011121314 committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Add Level-10 Class Command: Support GUI Class Main: Support FXML Class Reminerebot: Support GUI Class TaskList: Support GUI Class Ui: Support GUI
  Add Level-10 Class DialogBox: third iteration Class Main: third iteration
  Add Level-10 Class Reminderebot: second iteration Class Main: second iteration
  Add Level-10 Class Reminderebot: first iteration Class Launcher: Launch GUI Class Dialogbox: styles dialogbox for GUI Class Main: Build application Add images for user and Reminderebot Add controls for dialogbox

testing1234567891011121314 committed:
Add A-Varargs
Class TaskList: add method to print all tasks in tasklist

testing1234567891011121314 committed:
Add A-Assertions
Class Reminderebot:
- Assert Storage filepath exists

Class MainWindow:
- Assert Reminderebot is injected for main window
- Assert user input is not empty

These assumptions are required for the Reminderebot chatbot to function
as intended

testing1234567891011121314 committed:
Add A-CodeQuality
Class Parser: fix imports
Class MainWindow: fix syntax

testing1234567891011121314 committed:
Add A-Streams
Class Storage: Import BufferedWriter

testing1234567891011121314 committed:
Merge pull request #1 from testing1234567891011121314/branch-A-Assertions

Add A-Assertions

testing1234567891011121314 committed:
Merge pull request #2 from testing1234567891011121314/branch-A-CodeQuality

Add A-CodeQuality

testing1234567891011121314 committed:
Merge pull request #3 from testing1234567891011121314/branch-A-Streams

Add A-Streams

testing1234567891011121314 committed:
Merge branch 'master' of https://github.com/testing1234567891011121314/ip

* 'master' of https://github.com/testing1234567891011121314/ip:
  Add A-Streams Class Storage: Import BufferedWriter
  Add A-CodeQuality Class Parser: fix imports Class MainWindow: fix syntax
  Add A-Assertions Class Reminderebot: - Assert Storage filepath exists

testing1234567891011121314 committed:
Add A-CI
- Use Github Actions to setup Continuous Integration for Reminderebot.

testing1234567891011121314 committed:
Merge branch 'branch-A-CI'

* branch-A-CI:
  Add A-CI - Use Github Actions to setup Continuous Integration for Reminderebot.

testing1234567891011121314 committed:
Add BCD-Extension
Class HelpCommand: Create a help manual for users
Class Ui: Add text for help manual
Class Storage: Refactor code for readability
Class Parser: Refactor code for readability
Class Reminderebot: Fix bug for user to close GUI
Class MainWindow: Fix bug for user to close GUI
Enum Commands: Add Help Command

testing1234567891011121314 committed:
Merge branch 'branch-C-Help'

* branch-C-Help:
  Add BCD-Extension Class HelpCommand: Create a help manual for users Class Ui: Add text for help manual Class Storage: Refactor code for readability Class Parser: Refactor code for readability Class Reminderebot: Fix bug for user to close GUI Class MainWindow: Fix bug for user to close GUI Enum Commands: Add Help Command

testing1234567891011121314 committed:
Fix bug for Mark Command
Class Ui: fix bug where task not shown after marking

Mark command does not show task being marked to user.

Mark command should show task being marked for
easier readability of user.

Let's
- update the algorithm to show task being marked

testing1234567891011121314 committed:
Add A-BetterGui
MainWindow.fxml: Automatically resize elements

Some elements do not automatically follow the dimension of app when
resizing:
- TextField does not automatically resize horizontally
- Send button does not follow to the bottom right of the app
- ScrollPane does not automatically resize either horizontally and
vertically

Let's anchor the following nodes:
- Anchor TextField to the bottom while also automatically resizing
horizontally
- Anchor Send Button to the bottom right of the app
- Anchor ScrollPane to resize both horizontally and vertically

testing1234567891011121314 committed:
Add A-BetterGui
Class DialogBox: Add CSS for DialogBox
Class Mainwindow: Add CSS for MainWindow
Class Reminderebot: Add support for CSS

Let's improve the GUI by:
- Flip text bubbles for Reminderebot
- Add padding and margin
- Add borders
- Change style dynamically based on command

testing1234567891011121314 committed:
Merge branch 'branch-A-BetterGui'

* branch-A-BetterGui:
  Add A-BetterGui Class DialogBox: Add CSS for DialogBox Class Mainwindow: Add CSS for MainWindow Class Reminderebot: Add support for CSS
  Add A-BetterGui MainWindow.fxml: Automatically resize elements

testing1234567891011121314 committed:
Add A-Personality
Class Main: Set title and image for Reminderebot
Class MainWindow: Set profile image for Reminderebot

Let's add personality to Reminderebot by:
- Set the title of app to Reminderebot
- Set the profile image for Reminderebot

testing1234567891011121314 committed:
Merge branch 'branch-A-Personality'

* branch-A-Personality:
  Add A-Personality Class Main: Set title and image for Reminderebot Class MainWindow: Set profile image for Reminderebot

testing1234567891011121314 committed:
Add A-UserGuide
README: update all user features for easy reading
Ui.png: upload screenshot for display
Cleanup all classes to prepare for release

Let's,
- remove dead code from class Reminderebot
- remove dead code from class Ui
- remove dead code from class FindCommand
- fix minor bug to display error message in Gui when attempting to
create a Deadline or Event object with datetime in wrong format
- add support for showing goodbye message before exiting

testing1234567891011121314 committed:
Add A-UserGuide
README: update all user features for easy reading
Ui.png: upload screenshot for display
Cleanup all classes to prepare for release

Let's,
- remove dead code from class Reminderebot
- remove dead code from class Ui
- remove dead code from class FindCommand
- fix minor bug to display error message in Gui when attempting to
create a Deadline or Event object with datetime in wrong format
- add support for showing goodbye message before exiting

testing1234567891011121314 committed:
Merge branch 'master' of https://github.com/testing1234567891011121314/ip

testing1234567891011121314 committed:
Merge branch 'master' of https://github.com/testing1234567891011121314/ip

testing1234567891011121314 committed:
Merge branch 'master' of https://github.com/testing1234567891011121314/ip

testing1234567891011121314 committed:
Cleanup DeadlineCommand
Remove all unnecessary comments

testing1234567891011121314 committed:
Cleanup DeleteCommand
Remove all unnecessary comments

testing1234567891011121314 committed:
Cleanup EventCommand
Remove all unnecessary comments

testing1234567891011121314 committed:
Cleanup MarkCommand
Remove all unnecessary comments

testing1234567891011121314 committed:
Cleanup UnmarkCommand
Remove all unnecessary comments

testing1234567891011121314 committed:
Update README

testing1234567891011121314 committed:
Update README.md
- Add screenshot to github pages

testing1234567891011121314 committed:
Update README.md
- add relative path for screenshot

testing1234567891011121314 committed:
Update README
- change relative filepath

testing1234567891011121314 committed:
Update README
- change relative filepath for screenshot

testing1234567891011121314 committed:
Update README
